### PR TITLE
CB-13145: Create playservices version preference in config.xml	

### PR DESCRIPTION
**Pull request by audreyso**
Still need to add in tests

### What does this PR do?
CB-13145: Create playservices version preference in config.xml

### Checklist
- [X] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [X] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [X] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: src/cordova/plugin/util.js [hunk 1]
```diff
@@ -35,3 +40,39 @@ function saveToConfigXmlOn (config_json, options) {
...
+        .filter(function (variableName) {
+            // discard variables with default value
+            return !(pluginVariables[variableName] || opts.cli_variables[variableName]);
+        });
+
+    if (missingVariables.length) {
+        events.emit('verbose', 'Removing ' + pluginInfo.dir + ' because mandatory plugin variables were missing.');
+        shell.rm('-rf', pluginInfo.dir);
+        var msg = 'Variable(s) missing (use: --variable ' + missingVariables.join('=value --variable ') + '=value).';
+        return Q.reject(new CordovaError(msg));
```
Comment (stevengill) [inline src/cordova/plugin/util.js]: Change this to `throw new CordovaError(msg)`

Comment (stevengill) [reply src/cordova/plugin/util.js]: get rid of Q dep

**Comment by stevengill** [comment]
This has been merged. Can you close it

**Comment by adriano-di-giovanni** [comment]
@stevengill , looking at the [issue on JIRA](https://issues.apache.org/jira/browse/CB-13145) I find that the fix has been released with cordova@7.2.0 but I couldn't find any blog post on [cordova.apache.org](http://cordova.apache.org/blog/):
* mentioning the release of cordova 7.2.0
* mentioning CB-13145 on cordova 8.0.0

Also, I couldn't find the `PlayServicesVersion` preference in the docs.

Is it possibile to take advantage of the new preference?

Last question: the `PlayServicesVersion` preference has to be used by plugin developers or setting it my project's config.xml will be useless. Is it correct?

Thanks,
Adriano

**Comment by macdonst** [comment]
@adriano-di-giovanni the way it works is you can set a preference in config.xml or plugin.xml. The config.xml one over-rides the one in plugin.xml if they have the same name. 

Check out the push plugin: 

https://github.com/phonegap/phonegap-plugin-push/blob/master/plugin.xml#L47

I set a default FCM version but folks can over-ride it. Then the user can make all these versions match in config.xml.

**Comment by adriano-di-giovanni** [comment]
This way is up to the plugin developers to stick to a convention. Correct?

**Pull request closed without merging**